### DISCUSSION TITLE
Error during deserialization?

### DISCUSSION THREAD

**Discussion by ghost** [esensar/kotlinx-serialization-msgpack#91] [Q&A]
I don't know if this is how it should return, but when deserializing one of the classes, it returns the LinkedHashMap class. 
```kotlin 
fun main() {
    val msgPack = MsgPack
    val byteArray = msgPack.encodeToByteArray(RedisTestPacket())
    val decode = msgPack.decodeFromByteArray<Any>(byteArray)

    println(decode::class.simpleName)
}

@Serializable
data class RedisTestPacket(
    val message: String = "Hi msgpack"
)
```

Output:
```
LinkedHashMap
```

**Comment by esensar** [comment]
Hi!

By default, type information is not serialized. You must specify which type you expect when deserializing by passing generic parameter (in your case `Any` is passed, but `RedisTestPacket` should be to get the expected result). If you wish to also encode type information, you can check out extension types, or you can look into custom serializers.

I hope this helps! Let me know if you need more information.
